executor/simple.go

```diff
@@ -629,6 +630,27 @@ func (e *SimpleExec) executeStartTransactionReadOnlyWithTimestampBound(ctx conte
 	if err != nil {
 		return err
 	}
+	dom := domain.GetDomain(e.ctx)
+	m, err := dom.GetSnapshotMeta(e.ctx.GetSessionVars().TxnCtx.StartTS)
+	if err != nil {
+		return err
+	}
+	staleVer, err := m.GetSchemaVersion()
+	if err != nil {
+		return err
+	}
+	failpoint.Inject("mockStalenessTxnSchemaVer", func(val failpoint.Value) {
+		if val.(bool) {
+			staleVer = e.ctx.GetSessionVars().TxnCtx.SchemaVersion - 1
+		} else {
+			staleVer = e.ctx.GetSessionVars().TxnCtx.SchemaVersion
```

Comment:
Is `GetSchemaVersion` & `TxnCtx.SchemaVersion` different in unit testing? Or you can just check `_, ok := val.(bool); ok`...